doc/decisions/base_name.md

```diff
@@ -0,0 +1,33 @@
+# Base Name
+
+## Problem
+
+Both applications and configuration file formats might have arbitrary strings which need to be
+encoded within a key name part.
+
+## Constraints
+
+- keySetBaseName (key, keyBaseName (key)) should be a NOP
+
+## Assumptions
+
+## Considered Alternatives
+
+- restrict what `keySetBaseName` can accept is against:
+  - easy to use of API: applications with some key names would suddenly fail
```

Comment:
I do not understand your input here. `keySetBaseName` is the easiest API as it is a single function call. Other proposals were more involved, and as such they are not so easy to use. The argument has nothing to do with backwards compatibility.

Reply:
I have a problem with "applications with some key names would suddenly fail". Especially the word "suddenly".

Since we all agree, whatever we do between now and 1.0  won't be backwards compatible, this can only refer to applications written for the new API. But those wouldn't failed suddenly nor unexpectedly. If they would simply be buggy, since they improperly used the new API.

Applications written for the old API or written by someone who expects the new API to behave the same as the old one, should also be considered buggy, if they fail. There was a breaking change so the application should have been adapted to this change.

> `keySetBaseName` is the easiest API as it is a single function call. Other proposals were more involved, and as such they are not so easy to use.

That is a fair point, but is a totally different argument that what you have written here.

---

I think want you wanted to say here is

"This makes the API harder to use in some cases. Some applications would need additional logic around `keySetBaseName`."

I do not agree that this is enough of a reason, since most applications won't be affected, but this statement would at least make sense.